File: src/Field.java
public class Field {
    private FieldState fieldstate;
    private final int yPos;
    private final int xPos;

    public Field(int yPos, int xPos) {
        this.yPos = yPos;
        this.xPos = xPos;
    }

    public int getFieldY() {
        return yPos;
    }

    public int getFieldX() {
        return xPos;
    }

    public void setState(FieldState fieldState) {
        fieldstate = fieldState;
    }

    public FieldState getState() {
        return fieldstate;
    }
}

File: src/Field.kt
class Field {
    var state: FieldState
    val fieldY: Int
    val fieldX: Int

    constructor(yPos: Int, xPos: Int) {
        this.fieldY = yPos
        this.fieldX = xPos
        this.state = FieldState.EMPTYFIELD
    }

    constructor(yPos: Int, xPos: Int, fieldState: FieldState) {
        this.fieldY = yPos
        this.fieldX = xPos
        this.state = fieldState
    }
}
